crates/sui-types/src/message_envelope.rs

```diff
@@ -191,21 +199,35 @@ where
         self.auth_signature.epoch
     }
 
-    pub fn verify_signature(&self, committee: &Committee) -> SuiResult {
-        self.data.verify(Some(self.auth_sig().epoch))?;
+    pub fn verify_signature(
+        &self,
+        committee: &Committee,
+        verify_params: &VerifyParams,
+    ) -> SuiResult {
+        let mut verify_params = verify_params.clone();
+        verify_params.epoch = Some(self.auth_sig().epoch);
```

Comment:
This makes me really nervous, why cannot we use the epoch from the passed in param?
Seems a anti-pattern to me even we need it somehow. Could we find a better way

Reply:
well, the reason we can't is that we don't want to - this function previously did not take an epoch argument, and I don't want to change that behavior.

however, because other verify calls *do* take an epoch parameter, it was convenient to re-use it here.

But, I agree that this is pretty gross, let me try to find a better way.

Reply:
doesn't `committee` have epoch? or I guess I don't understand what you are discussing

Reply:
cleaned all this up with a better API